File: src/sendereCommons/Converters.java
package sendereCommons;

public class Converters {

    public static byte[] longToBytes(long l) {
        byte[] result = new byte[8];
        for (int i = 7; i >= 0; i--) {
            result[i] = (byte)(l & 0xFF);
            l >>= 8;
        }
        return result;
    }

    public static long bytesToLong(final byte[] b) {
        return bytesToLong(b,0);
    }

    public static long bytesToLong(final byte[] b, int offset) {
        long result = 0;
        for (int i = offset; i < 8+offset; i++) {
            result <<= 8;
            result |= (b[i] & 0xFF);
        }
        return result;
    }
}

File: src/sendereCommons/Converters.kt
package sendereCommons

object Converters {
    fun longToBytes(l: Long): ByteArray {
        var l = l
        val result = ByteArray(8)
        for (i in 7 downTo 0) {
            result[i] = (l and 0xFF).toByte()
            l = l shr 8
        }
        return result
    }

    @JvmOverloads
    fun bytesToLong(b: ByteArray, offset: Int = 0): Long {
        var result: Long = 0
        for (i in offset until 8 + offset) {
            result = result shl 8
            result = result or (b[i].toLong() and 0xFF)
        }
        return result
    }
}